Mathlib/Order/UpperLower/Basic.lean

```diff
@@ -511,15 +511,25 @@ section LE
 
 variable [LE α]
 
-/-- The type of upper sets of an order. -/
+/--
+The type of upper sets of an order.
+
+An upper set in an order is a subset such that any element greater than one of its members is
```

Comment:
I forget the actual syntax for it, but this might be one of the rare cases where `extend_doc` could be useful.  Maybe you can use `@[inherit_doc IsUpperSet]` on `UpperSet` and the `extend_doc UpperSet before "..."

Reply:
I just tested and this has the same effect as your change:
```lean
@[inherit_doc IsUpperSet]
structure UpperSet (α : Type*) [LE α] where
  /-- The carrier of an `UpperSet`. -/
  carrier : Set α
  /-- The carrier of an `UpperSet` is an upper set. -/
  upper' : IsUpperSet carrier
#align upper_set UpperSet

extend_docs UpperSet before "The type of upper sets of an order."

@[inherit_doc IsLowerSet]
structure LowerSet (α : Type*) [LE α] where
  /-- The carrier of a `LowerSet`. -/
  carrier : Set α
  /-- The carrier of a `LowerSet` is a lower set. -/
  lower' : IsLowerSet carrier
#align lower_set LowerSet

extend_docs LowerSet before "The type of lower sets of an order."
```

and moreover you do not have to worry about synchronising the doc-strings!

Reply:
>  this has the same effect as your change:

It doesn't because I cunningly removed a redundant `alpha` from the IsUpperSet docstring ;-) But I'm happy to put it back.